packages/generator/src/unique-name-finder.ts

```diff
@@ -0,0 +1,136 @@
+/* Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. */
+import {
+  reservedObjectPrototypeKeywords,
+  reservedVdmKeywords
+} from './name-formatting-reserved-key-words';
+
+type Separator = '-' | '_';
+
+export interface UniqueName {
+  uniqueName: string;
+  relatedUniqueNames: string[];
+}
+export class UniqueNameFinder {
```

Comment:
I think this the general concept here makes a lot of sense (especially in our case). When I look at where this class is used though, I wonder whether separating it from the according cache (alreadyUsedNames) is a good idea.
Every occurence of `.getInstance()` is followed by `.withAlreadyUsedNames` - maybe instead of having four caches and creating a new instance of this class for every new name, it would be more elegant to have four instances of this class, that have their own cache (which is already there) and are reused throughout the service formatter. This should not even be a big change.

Reply:
I thought the same but did not want to include extra changes. I adjusted the caches to be finders now.